pkg/scale/decode.go

```diff
@@ -655,61 +733,60 @@ func (ds *decodeState) decodeFixedWidthInt(dstv reflect.Value) (err error) {
 		var b byte
 		b, err = ds.ReadByte()
 		if err != nil {
-			return
+			return fmt.Errorf("reading byte: %w", err)
 		}
 		out = int8(b)
 	case uint8:
 		var b byte
 		b, err = ds.ReadByte()
 		if err != nil {
-			return
+			return fmt.Errorf("reading byte: %w", err)
 		}
 		out = b
 	case int16:
 		buf := make([]byte, 2)
 		_, err = ds.Read(buf)
 		if err != nil {
-			return
+			return fmt.Errorf("reading buffer: %w", err)
```

Comment:
nit mentioning the matching type would be nice

```suggestion
			return fmt.Errorf("reading buffer for int16: %w", err)
```